client/blocks/login/index.jsx

```diff
@@ -108,25 +109,37 @@ class Login extends Component {
 		const {
 			twoFactorAuthType,
 			twoFactorEnabled,
+			twoFactorNotificationSent,
 		} = this.props;
 
 		const {
 			rememberMe,
 		} = this.state;
 
+		let poller;
+		if ( twoFactorEnabled && twoFactorAuthType && twoFactorNotificationSent === 'push' ) {
+			poller = <PushNotificationApprovalPoller onSuccess={ this.rebootAfterLogin } />;
+		}
+
 		if ( twoFactorEnabled && includes( [ 'authenticator', 'sms', 'backup' ], twoFactorAuthType ) ) {
 			return (
-				<VerificationCodeForm
-					rememberMe={ rememberMe }
-					onSuccess={ this.rebootAfterLogin }
-					twoFactorAuthType={ twoFactorAuthType }
-				/>
+				<div>
+					{ poller }
```

Comment:
It doesn't get unmounted when navigating to a different 2FA screen. Pretty cool!